enterprise_learner_portal/api/v1/views.py

```diff
@@ -55,16 +42,50 @@ def get(self, request):
             user_id=user.id,
             enterprise_customer__uuid=enterprise_customer_id,
         )
-        course_ids_for_ent_enrollments = EnterpriseCourseEnrollment.objects.filter(
+        enterprise_enrollments = EnterpriseCourseEnrollment.objects.filter(
             enterprise_customer_user=enterprise_customer_user
-        ).values_list('course_id', flat=True)
-
-        overviews = get_course_overviews(course_ids_for_ent_enrollments)
+        )
 
         data = EnterpriseCourseEnrollmentSerializer(
-            overviews,
+            enterprise_enrollments,
             many=True,
             context={'request': request},
         ).data
 
         return Response(data)
+
+    def patch(self, request):
+        """
+        Patch method for the view.
+        """
+        user = request.user
+        enterprise_customer_id = request.query_params.get('enterprise_id', None)
+        course_id = request.query_params.get('course_id', None)
+        marked_done = request.query_params.get('marked_done', None)
+        if not (enterprise_customer_id and course_id) or marked_done is None:
```

Comment:
Personal preference, but I think I prefer not mixing ANDs and ORs if it's possible. Something like:

```
if not enterprise_customer_id or not course_id or marked_done is None:
```